pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -4486,6 +4522,7 @@ func TestScalingWithRules(t *testing.T) {
 		},
 	}
 
+	ctx := context.Background()
```

Comment:
```suggestion
	tCtx := ktesting.Init(t)
```

refer to https://github.com/kubernetes/kubernetes/pull/123399
> This simplifies test code by using the automatic context cancellation support in the updated ktesting package.

Reply:
Updated, thanks a lot!